docker-compose.ssl.dev.yml

```diff
@@ -46,8 +46,27 @@ services:
     ports:
       - "6767:6767"
 
+  elasticsearch:
+    image: elasticsearch:7.17.5
+    restart: always
+    ports:
+      - "9200:9200"
+      - "9300:9300"
+    environment:
+      - node.name=es-node01
+      - cluster.name=es-cluster
+      - discovery.type=single-node
+      - "ES_JAVA_OPTS=-Xms128m -Xmx128m"
```

Comment:
Isn't that too low for java?

Reply:
I had experience that indexer can fail over time if memory limits were low. Starts okay but then fails.

Reply:
> I had experience that indexer can fail over time if memory limits were low. Starts okay but then fails.

I'd expect as much. As far as I know standard java runtime takes about ≈50mb by itself, and with how heap is organized, I'd recommend to give it at least 512 mb.

Reply:
Also, no reason to set min heap equal to max heap. It's a premature optimization that will unnecessarily hog memory.

Reply:
changed to 512, and setting both values to it, I'm not an expert here and in all examples values are the same

Reply:
I'm an expert 8+ years java experience and I advise not to do it :)

https://stackoverflow.com/questions/43651167/is-there-any-advantage-in-setting-xms-and-xmx-to-the-same-value
We don't want to microoptimize the latency of the elasticsearch, and our memory is limited, so we better let JVM manage it.

Reply:
They [recommend](https://www.elastic.co/guide/en/elasticsearch/reference/6.6/heap-size.html) to use same value for both  and have it set up to 50% or server RAM (but not more than 32GB) but this is for dedicated host

Reply:
> They [recommend](https://www.elastic.co/guide/en/elasticsearch/reference/6.6/heap-size.html) to use same value for both and have it set up to 50% or server RAM (but not more than 32GB) but this is for dedicated host

That's for the case when the server's primary function is to host elasticsearch node. In this case it's reasonable to split memory between JVM and OS (file cache) and pre-allocate it to eliminate heap resizes. 

Our usecase is very different. Elasticsearch would be used only occasionally, and we don't really care about search latency. Having more memory available for the file cache (mysql performance) is more important for us.

Reply:
@Aivean I agree, could you please adjust these values in your PR that you are planning, thanks

Reply:
I discovered that elasticsearch explicitly checks that xms == xmx at startup and won't start if the are not equal. (sigh). Let's set them both to 512mb and see how that works.